pkg/controllers/dashboard/systemcharts/controller.go

```diff
@@ -204,22 +211,23 @@ func (h *handler) getChartsToInstall() []*chart.Definition {
 			},
 			Enabled: func() bool {
 				toEnable := false
-				suc, err := h.deployment.Get(namespace.System, sucDeploymentName)
+				suc, err := h.deploymentCache.Get(namespace.System, sucDeploymentName)
 
 				// the absence of the deployment or the absence of the legacy label on the existing deployment indicate
 				// that the old rancher-k3s/rke2-upgrader Project App has been removed
 				if err != nil {
-					logrus.Debugf("[systemcharts] failed to get the deployment %s/%s: %s", namespace.System, sucDeploymentName, err.Error())
 					if errors.IsNotFound(err) {
 						toEnable = true
+					} else {
+						logrus.Warnf("[systemcharts] failed to get the deployment %s/%s: %s", namespace.System, sucDeploymentName, err.Error())
 					}
 				}
 				if suc != nil {
 					if _, ok := suc.Labels[legacyAppLabel]; !ok {
 						toEnable = true
 					}
 				}
-				logrus.Debugf("[systemcharts] feature ManagedSystemUpgradeController = %t, toEnable = %t",
+				logrus.Infof("[systemcharts] feature ManagedSystemUpgradeController = %t, toEnable = %t",
```

Comment:
Any specific reason we're changing this from Debug to Info?

Reply:
Since debug-level logging isn’t set by default for the cluster-agent, we miss the initial logs if we enable debug logging afterward, as the existing pod will be replaced. Switching to info-level logging would make troubleshooting more straightforward.